tests/unit/query_plan.cpp

```diff
@@ -426,7 +426,54 @@ TYPED_TEST(TestPlanner, CreateWithSum) {
   AstStorage storage;
   auto ident_n = IDENT("n");
   auto n_prop = PROPERTY_LOOKUP(ident_n, prop);
-  auto sum = SUM(n_prop);
+  auto sum = SUM(n_prop, false);
+  auto query = QUERY(SINGLE_QUERY(CREATE(PATTERN(NODE("n"))), WITH(sum, AS("sum"))));
+  auto symbol_table = memgraph::query::MakeSymbolTable(query);
+  auto acc = ExpectAccumulate({symbol_table.at(*ident_n)});
+  auto aggr = ExpectAggregate({sum}, {});
+  auto planner = MakePlanner<TypeParam>(&dba, storage, symbol_table, query);
+  // We expect both the accumulation and aggregation because the part before
+  // WITH updates the database.
+  CheckPlan(planner.plan(), symbol_table, ExpectCreateNode(), acc, aggr, ExpectProduce());
+}
+
+TYPED_TEST(TestPlanner, MatchWithSumWithDistinctWhereReturn) {
+  // Test MATCH (n) WITH SUM(n.prop) + 42 AS sum WHERE sum < 42
```

Comment:
The comment is not in sync with the test in terms of `DISTINCT`: in the test we create a `SUM` operator with `true` as `distinct`, but in the query I don't see `DISTINCT`. Please add the `DISTINCT` keyword to the query in all test cases it is applicable and you added.